risc0/zkvm/src/host/server/exec/syscall/fork.rs

```diff
@@ -0,0 +1,333 @@
+            self.registers[idx] = data;
+        }
+        Ok(())
+    }
+
+    fn load_memory(&mut self, addr: WordAddr) -> Result<u32> {
+        let page_idx = addr.page_idx();
+        let idx = self.page_table[page_idx as usize];
+        let data = if idx == INVALID_IDX {
+            self.ctx.load_u32(addr.baddr())?
+        } else {
+            self.page_cache[idx as usize].load(addr)
+        };
+        // tracing::trace!("load_memory: {addr:?} -> 0x{data:08x}");
+        Ok(data)
+    }
+
+    fn store_memory(&mut self, addr: WordAddr, data: u32) -> Result<()> {
+        tracing::trace!("store_memory: {addr:?} <= 0x{data:08x}");
+        let page_idx = addr.page_idx();
+        let mut idx = self.page_table[page_idx as usize];
+        if idx == INVALID_IDX {
+            idx = self.page_cache.len() as u32;
+            tracing::trace!("load_page: {page_idx}, {idx}");
+            self.page_table[page_idx as usize] = idx;
+            self.page_cache.push(Page(self.ctx.load_page(page_idx)?));
+        }
+        self.page_cache[idx as usize].store(addr, data);
+
+        Ok(())
+    }
+
+    fn check_data_load(&self, addr: ByteAddr) -> bool {
+        is_guest_memory(addr.0)
+    }
+
+    fn check_data_store(&self, addr: ByteAddr) -> bool {
+        is_guest_memory(addr.0)
+    }
+
+    fn check_insn_load(&self, addr: ByteAddr) -> bool {
+        is_guest_memory(addr.0)
+    }
+}
+
+impl<'a, 'b> SyscallContext<'a> for ChildExecutor<'a, 'b> {
+    fn get_pc(&self) -> u32 {
+        self.pc.0
+    }
+
+    fn get_cycle(&self) -> u64 {
+        0
+    }
+
+    fn load_register(&mut self, idx: usize) -> u32 {
+        EmuContext::load_register(self, idx).unwrap()
+    }
+
+    fn load_u8(&mut self, addr: ByteAddr) -> Result<u8> {
+        ChildExecutor::load_u8(self, addr)
+    }
+
+    fn load_u32(&mut self, addr: ByteAddr) -> Result<u32> {
+        ChildExecutor::load_memory(self, addr.waddr())
+    }
+
+    fn load_page(&mut self, page_idx: u32) -> Result<Vec<u8>> {
+        let idx = self.page_table[page_idx as usize];
+        if idx == INVALID_IDX {
+            self.ctx.load_page(page_idx)
+        } else {
+            Ok(self.page_cache[idx as usize].0.clone())
+        }
+    }
+
+    fn syscall_table(&self) -> &SyscallTable<'a> {
+        &self.syscall_table
+    }
+}
+
+impl Page {
+    fn load(&self, addr: WordAddr) -> u32 {
+        let word_addr = (addr.0 % PAGE_WORDS as u32) as usize;
+        let byte_addr = word_addr * WORD_SIZE;
+        let mut bytes = [0u8; WORD_SIZE];
+        bytes.clone_from_slice(&self.0[byte_addr..byte_addr + WORD_SIZE]);
+        let data = u32::from_le_bytes(bytes);
+        tracing::trace!("load({addr:?}) -> 0x{data:08x}");
+        data
+    }
+
+    fn store(&mut self, addr: WordAddr, data: u32) {
+        let word_addr = (addr.0 % PAGE_WORDS as u32) as usize;
+        let byte_addr = word_addr * WORD_SIZE;
+        tracing::trace!("store({addr:?}, 0x{data:08x})");
```

Comment:
The formatting specifier `{data:#010x}` would also work like `0x{data:08x}`, but perhaps it's less clear.